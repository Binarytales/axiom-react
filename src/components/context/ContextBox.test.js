import React from 'react';
import renderer from 'react-test-renderer';
import { ContextBox } from 'bw-axiom';

function getComponent(props = {}) {
  return renderer.create(
    <ContextBox { ...props }>
      Lorem ipsum
    </ContextBox>
  );
}

describe('ContextBox', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders top', () => {
    const component = getComponent({ position: 'top' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders right', () => {
    const component = getComponent({ position: 'right' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders bottom', () => {
    const component = getComponent({ position: 'bottom' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders left', () => {
    const component = getComponent({ position: 'left' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders padding none', () => {
    const component = getComponent({ padding: 'none' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders padding small', () => {
    const component = getComponent({ padding: 'small' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders padding large', () => {
    const component = getComponent({ padding: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders custom width', () => {
    const component = getComponent({ width: '20rem' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
