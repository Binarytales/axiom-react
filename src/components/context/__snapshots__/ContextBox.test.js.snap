exports[`ContextBox renders bottom 1`] = `
<div
  className="ax-context-box ax-context-box--bottom ax-context-box--padded"
  style={
    Object {
      "width": "14.5rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;

exports[`ContextBox renders custom width 1`] = `
<div
  className="ax-context-box ax-context-box--top ax-context-box--padded"
  style={
    Object {
      "width": "20rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;

exports[`ContextBox renders left 1`] = `
<div
  className="ax-context-box ax-context-box--left ax-context-box--padded"
  style={
    Object {
      "width": "14.5rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;

exports[`ContextBox renders right 1`] = `
<div
  className="ax-context-box ax-context-box--right ax-context-box--padded"
  style={
    Object {
      "width": "14.5rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;

exports[`ContextBox renders top 1`] = `
<div
  className="ax-context-box ax-context-box--top ax-context-box--padded"
  style={
    Object {
      "width": "14.5rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;

exports[`ContextBox renders unpadded 1`] = `
<div
  className="ax-context-box ax-context-box--top"
  style={
    Object {
      "width": "14.5rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;

exports[`ContextBox renders with defaultProps 1`] = `
<div
  className="ax-context-box ax-context-box--top ax-context-box--padded"
  style={
    Object {
      "width": "14.5rem",
    }
  }>
  Lorem ipsum
  <span
    className="ax-context-box__arrow" />
</div>
`;
